core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -76,25 +86,70 @@ protected boolean requiresRelease() {
         return true;
     }
 
-    private ProbeResult fromSCMLink(ProbeContext context, String scm) {
+    private ProbeResult fromSCMLink(ProbeContext context, String scm, String pluginName) {
         Matcher matcher = GH_PATTERN.matcher(scm);
         if (!matcher.find()) {
             if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("{} is not respecting the SCM URL Template", scm);
+                LOGGER.debug("{} is not respecting the SCM URL Template.", scm);
             }
-            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories");
+            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories.");
         }
-
         try {
             context.getGitHub().getRepository(matcher.group("repo"));
-            return ProbeResult.success(key(), "The plugin SCM link is valid");
+            Optional<Path> pluginPathInRepository  = findPluginPom(context.getScmRepository(), pluginName);
+            String folderPath = pluginPathInRepository
+                .flatMap(path -> Optional.ofNullable(path.getParent()))
+                .map(parent -> parent.getFileName().toString())
+                .orElseGet(() -> pluginPathInRepository.map(Path::toString).orElse(""));
+            context.setScmFolderPath(folderPath);
+            return ProbeResult.success(key(), "The plugin SCM link is valid.");
         } catch (IOException ex) {
-            return ProbeResult.failure(key(), "The plugin SCM link is invalid");
+            return ProbeResult.failure(key(), "The plugin SCM link is invalid.");
         }
     }
 
     @Override
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom file in every directory available in the repository
+     *
+     * @param directory path in the scm
+     * @param pluginName the name of the plugin
+     * @return folderPath if it is valid
+     */
+        private Optional<Path> findPluginPom(Path directory, String pluginName) {
+        if (! Files.isDirectory(directory)) {
+            LOGGER.error("Directory {} does not exists during the probe during probe {}.", directory, pluginName);
+            return Optional.empty();
+        }
+
+        try (Stream<Path> paths = Files.find(directory, 3, (path, $) ->
+            "pom.xml".equals(path.getFileName().toString()))) {
+            return paths
+                .filter(pom -> pomFileMatchesPlugin(pom, pluginName))
+                .findFirst()
+                .or(() -> Optional.of(directory.getParent().getFileName()));
```

Comment:
In the previous reviews, you said that a `pom` file should always exist in the Maven plugin. Which is true. 

I assumed a pom file exist in the default parent directory. 

> Finding the path of the parent directory, either of the found pom.xml or the directory you were searching in, should be done elsewhere.

Do you mean when the `Optional.empty` is returned I should write the logic to look for its parent in another method?

Reply:
What I mean is that a function should return a consistent result.

Here, you either return a path that ends with `/pom.xml` if it is found, or you return a path that does not end by `/pom.xml` if the file is not found.

What I am saying is that it should be clear and consistent:

- Either you return the path ending with `/pom.xml` if you find it, and `empty` if you don't find the file (which you are right I said should be an error)
- Or you return the path that is supposed to contain the `pom.xml` inferring what it should be as a default case if the file is not found.

A mix of approaches does not make the code very clear.

I would personally advise to return the `/pom.xml` path if the file is found, and throw an error if the file is not found. The only case in which this will happen (according to Adrien) is for Gradle plugins, but I assume it's OK that the probe is broken in that case.

Reply:
I understand. You want me the method to do one thing at a time for better readability. 

I have followed your advise. Thanks.